src/public/js/app/progress_bars/traces/progress_bar_trace_filters_controller.js

```diff
@@ -0,0 +1,87 @@
+import { ProgressBarReader } from "/js/app/progress_bars/progress_bar_reader.js";
+import { SORT_OPTIONS } from "/js/app/progress_bars/traces/progress_bar_trace_sort.js";
+import { FormGroup } from "/js/app/resources/form_group.js";
+import { SearchController } from "/js/app/resources/search_controller.js";
+import { Observable } from "/js/lib/observable.js";
+
+export class ProgressBarTraceFiltersController {
+    /**
+     * 
+     * @param {import("/js/app/progress_bars/traces/progress_bar_trace_filters.js").ProgressBarTraceFilters} filters 
+     *   the filters to show and allow editing of
+     * @param {import("/js/app/progress_bars/traces/progress_bar_trace_sort.js").ProgressBarTraceSort} sort 
+     *   the sort to show and allow editing of
+     */
+    constructor(filters, sort) {
+        /**
+         * the element this view is attached to
+         * @type {Element}
+         * @readonly
+         */
+        this.element = document.createElement("div");
+        /**
+         * the filters to show and allow editing of
+         * @type {Observable.<import("/js/app/progress_bars/traces/progress_bar_trace_filters.js").ProgressBarTraceFilters>}
+         * @readonly
+         */
+        this.filters = new Observable(filters);
+        /**
+         * the sort to show and allow editing of
+         *  @type {Observable.<import("/js/app/progress_bars/traces/progress_bar_trace_sort.js").ProgressBarTraceSort>}
+         * @readonly
+         */
+        this.sort = new Observable(sort);
+        this.render();
+    }
+    render() {
+        this.element.classList.add("progress-bar-trace-filters-controller", "elevation-medium");
+        this.element.appendChild(new FormGroup((() => {
+            const searchController = new SearchController(
+                "name",
+                new ProgressBarReader(),
+                (f) => f.name,
+                (f, v) => Object.assign({}, f, { name: v })
+            );
+            searchController.value.addListener((progressBar) => {
+                if (progressBar !== null) {
+                    this.filters.value = Object.assign(
+                        {},
+                        this.filters.value,
+                        {
+                            progressBarName: { operator: "eq", value: progressBar.get("name") }
+                        }
+                    );
+                } else if (this.filters.value.progressBarName !== null) {
+                    this.filters.value = Object.assign(
```

Comment:
needs guard to prevent unnecessarily reloading the search items (causes page flicker)